### PR TITLE
petals models local support using python env

### PR DESCRIPTION
**Pull request by biswaroop1547**
- [x] depends on #393
- [x] depends on #440

Before merge, few issues that needs to be taken care of:

- [x] [Manage two execution variants of Stable Beluga](https://github.com/premAI-io/prem-registry/pull/88) (docker & python env local) through different `id` and other fields, in prem-registry. - We can remove the docker variant (https://github.com/premAI-io/prem-registry/pull/88)
- [x] stable beluga currently concatenates user prompt too in generated responses, make sure it only shows newly generated tokens. (https://github.com/premAI-io/prem-services/pull/125)
- [x] add prompt template for stable beluga (llama based ? or orca based ? need to check) (https://github.com/premAI-io/prem-services/pull/125)
- [x] [NOT CRITICAL] In UI the generated responses from Stable Beluga shows up as ~`strikethrough`~
image

- [x] ~Takes a bit of time when starting for first time (as env creation also happens if swarm-mode wasn't run before). Need to show some message here?~ (Not an issue currently)
- [x] [NOT CRITICAL] generation for `max_new_tokens > around 5` fails sometimes since it takes longer and variable time for generation and generally a timeout occurs from prem-app's side looks like (https://github.com/premAI-io/prem-services/pull/127): 
image

### CODE REVIEW

File: src-tauri/src/controller_binaries.rs [hunk 1]
```diff
@@ -88,6 +90,11 @@ pub async fn start_service(
             )
         })?
         .as_str();
+    let is_petals_model = services_guard
+        .get(&service_id)
+        .with_context(|| format!("service_id {} doesn't exist in registry", service_id))?
+        .petals
```
Comment (casperdcl) [inline src-tauri/src/controller_binaries.rs:96]: maybe ok for now, but long-term would [prefer `serviceType=python`](https://github.com/premAI-io/prem-registry/pull/88#issuecomment-1782613906)

**Comment by casperdcl** [comment]
just merged #440, lemme know if you need any help rebasing @biswaroop1547 @_)

**Comment by tiero** [comment]
Using[ latest commit ](https://github.com/premAI-io/prem-app/pull/435/commits/93488c49cb836f8c75ab347afc654e082fbe97b6) I downloaded Stable Beluga, once I click Open it hangs and seems to never really start

**Comment by biswaroop1547** [comment]
@tiero actually it takes around ~0.5-1min to startup the model server, to check if the server is up you can also do `lsof -i :8734`

**Comment by biswaroop1547** [comment]
- [ ] depends on https://github.com/premAI-io/prem-app/pull/477
- [x] depends on https://github.com/premAI-io/prem-app/pull/471

**Comment by tiero** [comment]
We should have a timeout, as I waited more than 5 minutes and it was keep haging to me. What can I do to debug?

**Comment by biswaroop1547** [comment]
@tiero that's weird because it shouldn't take more than 30 secs (given you've ran the swarm before, because that creates the python env which'll be reused for petals), but if you're starting anew then it'd take around 3 mins as it also installs and sets up the python environment before starting the server (currently when the python env is being setup after you click "open" we are not showing any message), to debug can you remove `~/.config/prem` once and then try again? (it'd take 2-3 mins for server to start)

**Comment by filopedraz** [comment]
The problem is not in the env creation. Attached here the logs.

image

**Comment by casperdcl** [comment]
I guess `dev` registry needs to be reverted back to `v1` before this can be merged, right?

**Comment by filopedraz** [comment]
Yes, correct @casperdcl. It's just for testing. The service actually works, but it takes a huge amount of time. After the health request was successful it took 60 seconds to load the chat screen.

image

**Comment by biswaroop1547** [comment]
@filopedraz yeah it takes longer if it's creating env from start, but if env is already present then it takes minimum 30 secs to maximum 1 min, do we want to show some kind of message when this is happening?

**Comment by filopedraz** [comment]
Good for now. I am more worried about the time between the toast and the load of the chat. I don't know what's. It happens to me with Mistral too actually.

**Comment by filopedraz** [comment]
Download doesn't even start now. Here a [loom](https://www.loom.com/share/ecb7d8ed481e42f28cd8383798b93ee0?sid=13d6b356-c780-4b7c-8148-de9e9b4b0f77).

**Comment by filopedraz** [comment]
Now it seems to work. I also rebased with `main`, but there seems to be an issue in the generation: 
image

**Comment by filopedraz** [comment]
Seems related to a special token. `it'`

**Comment by filopedraz** [comment]
The PR looks good and it works well for me. I created a new issue [here](https://github.com/premAI-io/prem-services/issues/139) for what concerns the generation.

**Review by filopedraz — APPROVED** [review]

**Comment by casperdcl** [comment]
I suggest we squash-merge because it's ultimately quite small & not worth rebasing/preserving history

**Pull request merged**